core/src/test/java/org/elasticsearch/index/query/FuzzyQueryBuilderTest.java

```diff
@@ -0,0 +1,145 @@
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.query.support.QueryParsers;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+public class FuzzyQueryBuilderTest extends BaseQueryTestCase<FuzzyQueryBuilder> {
+
+    private static final String[] REWRITE_METHOD = new String[]{"constant_score_auto", "scoring_boolean", "constant_score_boolean",
+            "constant_score_filter", "top_terms_boost_1", "top_terms_1"};
+
+    private static final String[] MAPPED_FIELD_NAME = new String[] { INT_FIELD_NAME, DOUBLE_FIELD_NAME, DATE_FIELD_NAME, STRING_FIELD_NAME };
+    
+    @Override
+    protected FuzzyQueryBuilder createTestQueryBuilder() {
+        FuzzyQueryBuilder query;
+
+        // mapped or unmapped field
+        if (randomBoolean()) {
+            Object text = randomAsciiOfLengthBetween(1, 50);
+            String fieldName = randomFrom(MAPPED_FIELD_NAME);
+            if (fieldName.equals(INT_FIELD_NAME)) {
+                text = randomIntBetween(0, 10);
+            } else if (fieldName.equals(DOUBLE_FIELD_NAME)) {
+                text = randomDouble() * 10;
+            } else if (fieldName.equals(DATE_FIELD_NAME)) {
+                text = new DateTime(System.currentTimeMillis() - randomIntBetween(0, 1000000), DateTimeZone.UTC);
+            }
+            query = new FuzzyQueryBuilder(fieldName, text);
+        } else {
+            query = new FuzzyQueryBuilder(randomAsciiOfLengthBetween(1, 10), randomAsciiOfLengthBetween(1, 50));
+        }
+        if (randomBoolean()) {
+            Fuzziness fuzziness;
+            switch (randomIntBetween(0, 2)) {
+                case 0:
+                    fuzziness = Fuzziness.fromEdits(randomIntBetween(0, 2));
+                    break;
+                case 1:
+                    fuzziness = Fuzziness.fromSimilarity(randomFloat());
+                    break;
+                default:
+                    fuzziness = Fuzziness.AUTO;
+            }
+            query.fuzziness(fuzziness);
+        }
+        if (randomBoolean()) {
+            query.prefixLength(randomIntBetween(0, 10));
+        }
+        if (randomBoolean()) {
+            query.maxExpansions(randomIntBetween(1, 10));
+        }
+        if (randomBoolean()) {
+            query.transpositions(randomBoolean());
+        }
+        if (randomBoolean()) {
+            query.rewrite(randomFrom(REWRITE_METHOD));
+        }
+        if (randomBoolean()) {
+            query.boost(2.0f / randomIntBetween(1, 20));
+        }
+        if (randomBoolean()) {
+            query.queryName(randomAsciiOfLengthBetween(1, 10));
+        }
+        return query;
+    }
+
+    @Override
+    protected Query createExpectedQuery(FuzzyQueryBuilder queryBuilder, QueryParseContext context) throws IOException {
```

Comment:
Much like with the tests for SimpleQueryStringQuery this method is almost a copy of the original toQuery code in the QueryBuilder. Can we track the reasoning behind this type of testing here: https://github.com/elastic/dev/blob/master/design/queries/general-guidelines.md (I believe we've discussed it in multiple PR threads and face2face but have lost of all the specific arguments).